bundles/org.openhab.binding.tado/src/main/java/org/openhab/binding/tado/internal/builder/AirConditioningZoneSettingsBuilder.java

```diff
@@ -60,10 +63,26 @@ public GenericZoneSetting build(ZoneStateProvider zoneStateProvider, GenericZone
             setting.setSwing(swing ? Power.ON : Power.OFF);
         }
 
+        if (light != null) {
+            setting.setLight(light ? Power.ON : Power.OFF);
+        }
+
         if (fanSpeed != null) {
             setting.setFanSpeed(getAcFanSpeed(fanSpeed));
         }
 
+        if (fanLevel != null) {
+            setting.setFanLevel(getFanLevel(fanLevel));
```

Comment:
Is `setFanLevel(null)` valid?

Reply:
> Is setFanLevel(null) valid?

No; that is why we need the null check on line 74; but I am not sure if I understand the point of your question..

PS see my above comments about null annotations. Please also note that some of the classes are JSON serialized classes, where the de-serializer is allowed to set the fields as null; so it is questionable whether it is better to have a) @NonNullByDefault on the class, and @Nullable on the fields, or b) declare the whole class without @NonNullByDefault annotation.

Reply:
See #12552

Reply:
>> Is setFanLevel(null) valid?
 
> No; that is why we need the null check on line 74; but I am not sure if I understand the point of your question..

I'm asking because `getFanLevel()` can return null even when not called with null as argument - at last line of the method. In this case you will call `setFanLevel()` also with null, since it's unchecked.

Reply:
Ok. I will check that again. And eventually make another commit tomorrow..

Reply:
> getFanLevel() can return null even when not called with null as argument - at last line of the method

Aha. I see what you mean. The last line of the method is necessary to satisfy the compiler warning that the method must have a `return` statement. But in fact the preceding `switch` statement is fully qualified so it is actually impossible for code execution to reach that line. So @jlaur I think that this solves your requirement to avoid the risk of NPE's. 

Nevertheless this case also underlines the need for proper null annotations and checks: In this PR I could refactor the `getFanLevel()` method in this PR (by adding a `default` case to the `switch` statement). But if you look at the other `getXXX()` methods in (say) this class, you can see that those methods also are guilty of the same crime, and would therefore also need to be properly refactored as well. As said before, I prefer to do that in a separate PR.